modules/zuora/src/zuoraClient.ts

```
@@ -93,23 +132,61 @@ export class ZuoraClient {
 			},
 			body,
 		});
-		const json = await response.json();
+		const responseBody = await response.text();
 
-		// Check both HTTP status and logical success
-		// Some Zuora endpoints return HTTP 200 with success: false for logical errors
-		const isHttpSuccess = response.ok;
+		const responseHeaders: Record<string, string> = Object.fromEntries(
+			[...response.headers.entries()].map(([k, v]) => [k.toLowerCase(), v]),
+		);
+		const result: RestResult = {
+			status: response.status,
+			responseBody,
+			responseHeaders,
+		};
+		if (!response.ok) {
+			throw RestClientError.create('http call failed', result);
+		}
 
-		if (isHttpSuccess && isLogicalSuccess(json)) {
+		try {
+			const json: unknown = JSON.parse(result.responseBody);
```

this is the main change:
Previously it would have returned a failed promise with an unhelpful message.

Now we parse the json inside a try block, meaning that if it fails, we already have the body as text, so we can throw an error containing what was actually in the body.